contacts/views.py

```diff
@@ -1,230 +1,157 @@
-@login_required
-def edit(request, id):
-    contact = Contact.objects.get(id=id)
-    if contact is not None:
-        return render(request, 'contacts/edit_contact.html', {'contact': contact, 'contact_id': id})
-
-
-@require_http_methods(["GET"])
-@login_required
-def delete(request, id):
-    contact = Contact.objects.get(id=id)
-    if contact is not None:
-        emails = Email.objects.filter(contact_id=id)
-        numbers = Number.objects.filter(contact_id=id)
-        if emails is not None and numbers is not None:
-            emails.delete()
-            numbers.delete()
-            contact.delete()
-            return HttpResponseRedirect('/user_index/')
-    return HttpResponse("Not delete id issue while try delete")
-
-
-def authenticate_user(request):
-    user_str = str(request.user)
-    if request.user is not None:
-        return HttpResponseRedirect('/user_index/')
+from django.utils.decorators import method_decorator
+from django.views.decorators.csrf import csrf_exempt
+from rest_framework import permissions, status
+from rest_framework.authtoken.models import Token
+from rest_framework.decorators import api_view, permission_classes
+from rest_framework.permissions import AllowAny
+from rest_framework.status import (
+    HTTP_400_BAD_REQUEST,
+    HTTP_404_NOT_FOUND,
+    HTTP_200_OK
+)
+from rest_framework.response import Response
+from rest_framework.views import APIView
+import datetime
+from contacts.serializers import UserSerializer, ContactSerializer, EmailSerializer, UserDetailSerializer, \
+    ContactDetailSerializer
+from contacts.models import User, Contact, Email
+from django.contrib.auth import authenticate
+
+
+class UserProfile(APIView):
+    permission_classes = (permissions.AllowAny,)
+
+    def get(self, request):
+        if 'token' in request.COOKIES:
+            pk = request.COOKIES.get('user_id')
+            user = User.objects.get(id=pk)
+            contacts = self.get_serialized(pk, user)
+            serializer = UserSerializer(user)
+            return Response(serializer.data, status=HTTP_200_OK)
+        else:
+            return Response("Your token expire Login again", status=HTTP_400_BAD_REQUEST)
+
+    def put(self, request):
+        if 'token' in request.COOKIES:
+            pk = request.COOKIES.get('user_id')
+            user = User.objects.get(id=pk)
+            payload = request.data
+            serializer = UserSerializer(user, data=payload, partial=True)
+            if serializer.is_valid(raise_exception=True):
+                serializer.save()
+            return Response(serializer.data, status=HTTP_200_OK)
+        else:
+            return Response("PUT REQUEST FAILED", status=HTTP_400_BAD_REQUEST)
+
+    def get_serialized(self, pk, user):
+
+        queryset = Contact.objects.filter(user_id=pk)
+
+        serializer = ContactSerializer(queryset, many=True)
+        data = serializer.data
+        return data
+
+
+class UserContacts(APIView):
```

Comment:
Separate API views in to an `api.py` file

Reply:
ok